File: src/main/java/com/brainstormer/controller/UserController.java
package com.brainstormer.controller;

import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.RestController;
import com.brainstormer.entity.User;
import com.brainstormer.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;


@RestController
@PreAuthorize("hasRole('ADMIN')")
public class UserController {

    private final UserRepository userRepository;

    @Autowired
    public UserController(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    @RequestMapping("/users")
    public Iterable<User> getUsers() {
        return userRepository.findAll();
    }

}

File: src/main/kotlin/com/brainstormer/controller/UserController.kt
package com.brainstormer.controller

import com.brainstormer.entity.User
import com.brainstormer.repository.UserRepository
import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@PreAuthorize("hasRole('ADMIN')")
class UserController (val repository:UserRepository) {

    @GetMapping("/users")
    fun findAll(): List<User>? = repository.findAll()
}